src/components/structures/TimelinePanel.js

```diff
@@ -510,8 +510,10 @@ var TimelinePanel = React.createClass({
         var currentReadUpToEventId = this._getCurrentReadReceipt(true);
         var currentReadUpToEventIndex = this._indexForEventId(currentReadUpToEventId);
 
+        currentReadUpToEventIndex = currentReadUpToEventIndex ||
```

Comment:
needs a comment to explain what we're doing.

Isn't zero a valid index? why do we fall back in that case?